src/libraries/System.Console/src/System/ConsolePal.iOS.cs

```diff
@@ -52,9 +52,9 @@ internal static void EnsureConsoleInitialized()
 
         internal static TextReader GetOrCreateReader() => throw new PlatformNotSupportedException();
 
-        public static bool NumberLock => false;
+        public static bool NumberLock => throw new PlatformNotSupportedException();
```

Comment:
Aside, there are now >150 instances of `throw new PlatformNotSupportedException();` in System.Console, would it save [significant codegen ](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNMAfAAgJgIwCwAUFgMwAEu5AwuQN7HlOUVYAs5AsgBQCU9jZkJgALBAHsA7uQB2caQAUANhBgAzcQjAA5cTADKAVwAOxzfAAmAUQAeAYzjGYAS3Ey+AbkFMAvt/L+ZJQc2nz+DERCQjy8XpHMfkQ+QA==) if these called a helper method?

Reply:
I'd be surprised if it would be noticeable. PNSE is thrown on platforms which run illinker so unless someone calls into these methods they will be removed.

Reply:
Ah - you're right. Only a few of them are in the Unix one. Fair enough.